src/pages/index.js

```diff
@@ -129,18 +117,43 @@ function getCardElement(data) {
 
   // -----Like Cards Feature-----
 
-  const cardLikeButton = cardElement.querySelector("#like-button");
+  const cardLikeButton = cardElement.querySelector(".card__like-button");
+
+  if (data.isLiked) {
+    cardLikeButton.classList.add("card__like-button_liked");
+  }
 
   cardLikeButton.addEventListener("click", () => {
-    cardLikeButton.classList.toggle("card__like-button_liked");
+    const isCurrentlyLiked = cardLikeButton.classList.contains(
+      "card__like-button_liked"
+    );
+    if (isCurrentlyLiked) {
+      api
+        .dislikeCard(data._id)
+        .then((updatedCard) => {
+          cardLikeButton.classList.remove("card__like-button_liked");
+        })
+        .catch((err) => {
+          console.error(err);
+        });
+    } else {
+      api
+        .likeCard(data._id)
+        .then((updatedCard) => {
+          cardLikeButton.classList.add("card__like-button_liked");
+        })
+        .catch((err) => {
+          cardLikeButton.classList.toggle("card__like-button_liked");
```

Comment:
`          cardLikeButton.classList.toggle("card__like-button_liked");
`

Please, delete it from `.catch`. You shouldn't change the color in case of an error


```js

        .catch((err) => {
          console.error("Error updating like status:", err);
        });
```